src/server.cpp

```diff
@@ -228,18 +228,32 @@ Server::Server(
 	m_nodedef(createNodeDefManager()),
 	m_craftdef(createCraftDefManager()),
 	m_thread(new ServerThread(this)),
-	m_uptime(0),
 	m_clients(m_con),
 	m_admin_chat(iface),
 	m_modchannel_mgr(new ModChannelMgr())
 {
-	m_lag = g_settings->getFloat("dedicated_server_step");
-
 	if (m_path_world.empty())
 		throw ServerError("Supplied empty world path");
 
 	if (!gamespec.isValid())
 		throw ServerError("Supplied invalid gamespec");
+
+#if USE_PROMETHEUS
+	m_metrics_backend = std::unique_ptr<MetricsBackend>(new PrometheusMetricsBackend());
+#else
+	m_metrics_backend = std::unique_ptr<MetricsBackend>(new MetricsBackend());
+#endif
+
+	m_uptime_counter = m_metrics_backend->addCounter("minetest_core_server_uptime", "Server uptime (in seconds)");
+	m_player_gauge = m_metrics_backend->addGauge("minetest_core_player_number", "Number of connected players");
+
+	m_timeofday_gauge = m_metrics_backend->addGauge("minetest_core_timeofday", "Time of day value");
+	m_lag_gauge = m_metrics_backend->addGauge("minetest_core_latency", "Latency value (in seconds)");
+	m_aom_buffer_counter = m_metrics_backend->addCounter("minetest_core_aom_buffer_count", "Active Object Message buffer queue size");
```

Comment:
line length

"queue" is a bit misleading, "Active Object Messages sent" would be more accurate
also, where is this value reset?

Reply:
okay for the typo, i don't understand your second question, can you be more accurate?

Reply:
it's a counter, no value reset if that's the question. Like any counter in metrology it ensure that you derive it on your graph side to properly see the rate of the messages over time and not miss the value pikes.
https://prometheus.io/docs/concepts/metric_types/
Here it's more a message rate, then counter is required